backtesting/backtesting.py

```diff
@@ -1224,101 +1244,151 @@ def maximize(stats: pd.Series, _key=maximize):
+            finally:
+                del Backtest._mp_backtests[backtest_uuid]
+
+            best_params = heatmap.idxmax()
+
+            if pd.isnull(best_params):
+                # No trade was made in any of the runs. Just make a random
+                # run so we get some, if empty, results
+                self.run(**param_combos[0])  # type: ignore
             else:
-                if os.name == 'posix':
-                    warnings.warn("For multiprocessing support in `Backtest.optimize()` "
-                                  "set multiprocessing start method to 'fork'.")
-                for batch_index in _tqdm(range(len(param_batches))):
-                    _, values = Backtest._mp_task(backtest_uuid, batch_index)
-                    for value, params in zip(values, param_batches[batch_index]):
-                        heatmap[tuple(params.values())] = value
-        finally:
-            del Backtest._mp_backtests[backtest_uuid]
-
-        best_params = heatmap.idxmax()
-
-        if pd.isnull(best_params):
-            # No trade was made in any of the runs. Just make a random
-            # run so we get some, if empty, results
-            self.run(**param_combos[0])  # type: ignore
+                # Re-run best strategy so that the next .plot() call will render it
+                self.run(**dict(zip(heatmap.index.names, best_params)))
+
+            if return_heatmap:
+                return self._results, heatmap
+            return self._results
+
+        def _optimize_skopt() -> Union[pd.Series, Tuple[pd.Series, pd.Series]]:
+
+            dimensions = []
+            for key, value in kwargs.items():
+                args_array = np.array(value)
+
+                if args_array.dtype.kind in 'ium':
+                    dimensions.append(Integer(low=min(value), high=max(value), name=key))
+                elif args_array.dtype.kind == 'f':
+                    dimensions.append(Real(low=min(value), high=max(value), name=key))
+                else:
+                    dimensions.append(Categorical(args_array.tolist(), name=key))
+
+            @use_named_args(dimensions=dimensions)
+            def convert(**params):
+                res = self.run(**params)
+                return -res[maximize_key]
+
+            res = forest_minimize(func=convert,
+                                  dimensions=dimensions,
+                                  n_calls=max_tries, base_estimator="ET",
+                                  random_state=4)
+
+            self.run(**dict(zip(list(kwargs.keys()), res.x)))
+
+            heatmap_df = pd.DataFrame(res.x_iters, columns=list(kwargs.keys()))
+            heatmap = pd.Series(res.func_vals,
+                                name=maximize_key,
+                                index=pd.MultiIndex.from_frame(heatmap_df))
+
+            if return_optimization and return_heatmap:
+                return self._results, res, heatmap
```

Comment:
```
Expected type 'Union[Series, Tuple[Series, Series]]', got 'Tuple[Any, Any, Series]' instead 
```
Just a small type hinting issue

Reply:
It's now `Tuple[Series, Series, scipy.optimize.OptimizeResult]`. Optimize result comes after heatmap. :ok_hand: